runtime/src/accounts_db.rs

```diff
@@ -5468,6 +5469,13 @@ impl AccountsDb {
             stats
                 .accounts_in_roots_older_than_epoch
                 .fetch_add(num_accounts, Ordering::Relaxed);
+            let ancients = sub_storages
+                .iter()
+                .map(|storage| if is_ancient(&storage.accounts) { 1 } else { 0 })
+                .sum();
```

Comment:
Maybe this simplification?
```suggestion
            let ancients = sub_storages
                .iter()
                .filter(is_ancient)
                .count();
```
(Note: I didn't try compiling this to ensure it actually builds 😅. May need an explicit type parameter on the `count()` or `ancients`.)

Reply:
see my next attempt